src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-sync-pipe-modal/rgw-multisite-sync-pipe-modal.component.ts

```diff
@@ -119,20 +124,47 @@ export class RgwMultisiteSyncPipeModalComponent implements OnInit {
       this.pipeForm.setErrors({ cdSubmitButton: true });
       return;
     }
-    this.rgwMultisiteService.createEditSyncPipe(this.pipeForm.getRawValue()).subscribe(
-      () => {
-        const action = this.editing ? 'Modified' : 'Created';
-        this.notificationService.show(
-          NotificationType.success,
-          $localize`${action} Sync Pipe '${this.pipeForm.getValue('pipe_id')}'`
-        );
-        this.activeModal.close('success');
-      },
-      () => {
-        // Reset the 'Submit' button.
-        this.pipeForm.setErrors({ cdSubmitButton: true });
-        this.activeModal.dismiss();
-      }
-    );
+
+    const sourceZones: any = { added: [], removed: [] };
+    const destZones: any = { added: [], removed: [] };
+    if (this.editing) {
+      destZones.removed = this.pipeSelectedRow.dest.zones.filter(
+        (zone: string) => !this.destZones.data.selected.includes(zone)
+      );
+      destZones.added = this.destZones.data.selected.filter(
+        (zone: string) => !this.pipeSelectedRow.dest.zones.includes(zone)
+      );
+      sourceZones.removed = this.pipeSelectedRow.source.zones.filter(
+        (zone: string) => !this.sourceZones.data.selected.includes(zone)
+      );
+      sourceZones.added = this.sourceZones.data.selected.filter(
+        (zone: string) => !this.pipeSelectedRow.source.zones.includes(zone)
```

Comment:
This is repeating so better to have a function which will give items to be added and removed.

Reply:
Thanks, did that